design/Soft_finality_guarantee.md

```diff
@@ -0,0 +1,161 @@
+factors unlikely to happen unless there is a re-org attack.
+
+
+## Outcomes of the design
+
+- In the context of fast finality and bridge designs, define the "Soft finality" guarantee.
+- Define the challenge conditions for the sequencer generating competing light batches.
+- Based on the above define the structure of a Light batch.
+
+## Assumption
+
+Messages from the L1 are processed as they are found in the L1 blocks, and synthetic transactions are generated based on them.
+These synthetic transactions are implicitly included in the light batches.
+
+
+## Solutions
+
+Since the problem is expressed in terms of "tension", we'll first try to explore the different options, and
+understand where they sit on the tradeoff spectrum.
+
+
+### Option 1 - Adding delays
+
+The straightforward solution is to wait a while before processing deposits so that these transactions are "final" on the L1.
+Eventually, the cost of mounting a re-org attack on the L1 becomes prohibitively high, ensuring security.
+
+There are two main disadvantages.
+
+The first is that users must wait before bridge messages are processed, which is not a great UX and it disables certain applications. 
+The second is that the solution is only as secure as the delay.
+
+The subtlety of this approach is that it can transform Obscuro into a side-chain even though we roll up to Ethereum.
+
+This depends on the definition of an L2, which is a moving target. If the consensus is that an L2 needs a re-org resistant
+bridge, then this option will break it.
+
+On the tradeoff spectrum, this solution lies towards the ideal: "The result of executing this transaction can never result in a different outcome" end,
+which is good, but it has some severe disadvantages.
+
+The Sequencer is never allowed to produce a competing sibling Light Batch. 
+
+
+### Option 2 - Link light batches to blocks
+
+Like POBI, each LB is generated from a single L1 block and links to it. (Actually, multiple LBs will be linked to a single block)
+
+The sequencer monitors the L1, and the moment a reorg is happening, it will produce new LBs linked to the blocks from 
+the new branch, which will then be distributed.
+
+The difference from POBI is that LBs are not published to a management contract in the L1, so they are no longer rejected 
+by a "source of truth" authority the moment they are no longer linked to a block found on the canonical chain. 
+They are only distributed from the sequencer to the other L2 nodes.
+This can lead to chaotic results and hard-to-enforce behaviour. Basically, the sequencer could use the re-org reason to generate competing LBs,
+and attempt some MEV, without being punished.
+
+From a UX pov, the soft finality guarantee is not great. If there is an L1 re-org, then the result of a transaction that
+has received a response already could change.
+
+
+### Option 3 - Link light batches to blocks, but guarantee the L2 ordering
```

Comment:
I think for this option, we need to explain how you discover the sequencer has cheated, and how you punish them.